lib/model/rwfolder.go

```diff
@@ -1372,12 +1374,49 @@ func (f *rwFolder) dbUpdaterRoutine() {
 	tick := time.NewTicker(maxBatchTime)
 	defer tick.Stop()
 
+	var changedFiles []string
+	var changedDirs []string
+	if f.fsync {
+		changedFiles = make([]string, 0, maxBatchSize)
+		changedDirs = make([]string, 0, maxBatchSize)
+	}
+
+	syncFilesOnce := func(files []string, syncFn func(string) error) {
+		sort.Strings(files)
+		var lastFile string
+		for _, file := range files {
+			if lastFile != file {
+				lastFile = file
+				if err := syncFn(file); err != nil {
+					l.Infoln("fsync failed:", err)
+				}
+			}
+		}
+	}
+
 	handleBatch := func() {
 		found := false
 		var lastFile protocol.FileInfo
 
 		for _, job := range batch {
 			files = append(files, job.file)
+			if f.fsync {
+				// collect changed files and dirs
+				switch job.jobType {
+				case dbUpdateShortcutFile:
+					changedFiles = append(changedFiles, filepath.Join(f.dir, job.file.Name))
+				case dbUpdateHandleFile:
+					changedFiles = append(changedFiles, filepath.Join(f.dir, job.file.Name))
+					fallthrough
+				case dbUpdateDeleteDir:
+					fallthrough
+				case dbUpdateDeleteFile:
+					changedDirs = append(changedDirs,
+						filepath.Dir(filepath.Join(f.dir, job.file.Name)))
```

Comment:
The line break is a bit ugly here.
I'd store the value in a variable or just leave one longer line.

Personally, if I do have to break brackets across multiple lines, I usually follow the same principle as with curly braces such as:

```
foo = append(
  bar_long_baz_nuff
)
```

argument being that nobody does:

```
if (x) {
    do_y() }
```

Yet that's just me.